src/Microsoft.AspNetCore.Mvc.Core/Internal/VirtualFileResultExecutor.cs

```diff
@@ -31,48 +32,87 @@ public VirtualFileResultExecutor(ILoggerFactory loggerFactory, IHostingEnvironme
 
         public Task ExecuteAsync(ActionContext context, VirtualFileResult result)
         {
-            SetHeadersAndLog(context, result);
-            return WriteFileAsync(context, result);
+            var fileInfo = GetFileInformation(result);
+            if (fileInfo.Exists)
+            {
+                var lastModified = result.LastModified == null ? fileInfo.LastModified : result.LastModified;
+                var (range, rangeLength, serveBody) = SetHeadersAndLog(
+                    context,
+                    result,
+                    fileInfo.Length,
+                    lastModified,
+                    result.EntityTag);
+                if (serveBody)
+                {
+                    return WriteFileAsync(context, result, fileInfo, range, rangeLength);
+                }
+            }
+            else
```

Comment:
That was not the behavior before adding the range. `FileInfo` was only checked in the `WriteFileAsync` method in `VirtualFileResultExecutor`